processing/src/main/java/io/druid/query/aggregation/cardinality/CardinalityAggregator.java

```diff
@@ -32,48 +32,58 @@
 public class CardinalityAggregator implements Aggregator
 {
   private final String name;
-  private final List<ColumnSelectorPlus<CardinalityAggregatorColumnSelectorStrategy>> selectorPlusList;
+  private final ColumnSelectorPlus<CardinalityAggregatorColumnSelectorStrategy>[] selectorPluses;
```

Comment:
In some places, you replace a selectorPlusList with an array (for performance?).  Would it make sense to make the same change elsewhere?  E.g., it looks like CardinalityAggregatorFactory specifically converts an array to a List, then passes the List to CardinalityAggregator, which just copies the List into a new array.